codegen-core/src/main/kotlin/software/amazon/smithy/rust/codegen/core/smithy/CodegenDelegator.kt

```diff
@@ -162,6 +162,12 @@ open class RustCrate(
         return this
     }
 
+    /**
+     * Returns the module for a given Shape.
+     */
+    fun moduleFor(shape: Shape, moduleWriter: Writable): RustCrate =
+        withModule((symbolProvider as RustSymbolProvider).moduleForShape(shape), moduleWriter)
+
```

Comment:
I'll look into the naming here. I think `RustCrate` is named after the current crate being rendered, but it's unclear to me why it is also the type representing modules.

Reply:
Oh, it returns `this` so that it can be function chained. It doesn't look like it actually ever is function chained though, so we could remove the return value if it's confusing.

Reply:
I'm fine with leaving this for now. We can think about updating `RustCrate` whenever we look at removing `RuntimeType`.